From 9039c943dca8044bf9d2d5e71213292ca1a21c9a Mon Sep 17 00:00:00 2001
From: Christophe Fergeau <christophe@fergeau.eu>
Date: Sun, 29 Mar 2026 20:00:36 +0200
Subject: [PATCH] Hacky patch to allow building offline

generate-release-data.js needs networking to fetch a release-data.json
file. This commit hardcodes it so that offline builds can be done.
---
 scripts/generate-release-data.js |  27 ---
 src/data/release-data.json       | 307 +++++++++++++++++++++++++++++++
 2 files changed, 307 insertions(+), 27 deletions(-)
 create mode 100644 src/data/release-data.json

diff --git a/scripts/generate-release-data.js b/scripts/generate-release-data.js
index 9349c35..4067608 100644
--- a/scripts/generate-release-data.js
+++ b/scripts/generate-release-data.js
@@ -6,34 +6,7 @@ const __filename = url.fileURLToPath(import.meta.url);
 const __dirname = path.resolve(__filename, "../..");
 
 async function generateReleaseData() {
-  try {
-    const response = await fetch(
-      "https://api.github.com/repos/zmkfirmware/zmk-studio/releases/latest",
-      {
-        headers: process.env.GITHUB_TOKEN
-          ? { Authorization: `Bearer ${process.env.GITHUB_TOKEN}` }
-          : {},
-      },
-    );
-    if (!response.ok) {
-      throw new Error(`HTTP error! status: ${response.status}`);
-    }
-
-    const data = await response.json();
-    const dataFilePath = path.resolve(
-      __dirname,
-      "src",
-      "data",
-      "release-data.json",
-    );
-    await fs.mkdir(path.dirname(dataFilePath), { recursive: true });
-    await fs.writeFile(dataFilePath, JSON.stringify(data));
-
     console.log("Release data generated successfully!");
-  } catch (error) {
-    console.error("Error generating release data:", error);
-    process.exit(1);
-  }
 }
 
 generateReleaseData();
diff --git a/src/data/release-data.json b/src/data/release-data.json
new file mode 100644
index 0000000..457bf13
--- /dev/null
+++ b/src/data/release-data.json
@@ -0,0 +1,307 @@
+{
+  "url": "https://api.github.com/repos/zmkfirmware/zmk-studio/releases/195729513",
+  "assets_url": "https://api.github.com/repos/zmkfirmware/zmk-studio/releases/195729513/assets",
+  "upload_url": "https://uploads.github.com/repos/zmkfirmware/zmk-studio/releases/195729513/assets{?name,label}",
+  "html_url": "https://github.com/zmkfirmware/zmk-studio/releases/tag/v0.3.1",
+  "id": 195729513,
+  "author": {
+    "login": "petejohanson",
+    "id": 202695,
+    "node_id": "MDQ6VXNlcjIwMjY5NQ==",
+    "avatar_url": "https://avatars.githubusercontent.com/u/202695?v=4",
+    "gravatar_id": "",
+    "url": "https://api.github.com/users/petejohanson",
+    "html_url": "https://github.com/petejohanson",
+    "followers_url": "https://api.github.com/users/petejohanson/followers",
+    "following_url": "https://api.github.com/users/petejohanson/following{/other_user}",
+    "gists_url": "https://api.github.com/users/petejohanson/gists{/gist_id}",
+    "starred_url": "https://api.github.com/users/petejohanson/starred{/owner}{/repo}",
+    "subscriptions_url": "https://api.github.com/users/petejohanson/subscriptions",
+    "organizations_url": "https://api.github.com/users/petejohanson/orgs",
+    "repos_url": "https://api.github.com/users/petejohanson/repos",
+    "events_url": "https://api.github.com/users/petejohanson/events{/privacy}",
+    "received_events_url": "https://api.github.com/users/petejohanson/received_events",
+    "type": "User",
+    "user_view_type": "public",
+    "site_admin": false
+  },
+  "node_id": "RE_kwDOL8zQ3M4Lqphp",
+  "tag_name": "v0.3.1",
+  "target_commitish": "b5888d9c86bd2fdc50212ad323779dad402ba50b",
+  "name": "v0.3.1",
+  "draft": false,
+  "immutable": false,
+  "prerelease": false,
+  "created_at": "2025-01-20T22:39:06Z",
+  "updated_at": "2025-01-20T22:43:47Z",
+  "published_at": "2025-01-20T22:39:18Z",
+  "assets": [
+    {
+      "url": "https://api.github.com/repos/zmkfirmware/zmk-studio/releases/assets/221951336",
+      "id": 221951336,
+      "node_id": "RA_kwDOL8zQ3M4NOrVo",
+      "name": "ZMK.Studio-0.3.1-1.x86_64.rpm",
+      "label": "",
+      "uploader": {
+        "login": "github-actions[bot]",
+        "id": 41898282,
+        "node_id": "MDM6Qm90NDE4OTgyODI=",
+        "avatar_url": "https://avatars.githubusercontent.com/in/15368?v=4",
+        "gravatar_id": "",
+        "url": "https://api.github.com/users/github-actions%5Bbot%5D",
+        "html_url": "https://github.com/apps/github-actions",
+        "followers_url": "https://api.github.com/users/github-actions%5Bbot%5D/followers",
+        "following_url": "https://api.github.com/users/github-actions%5Bbot%5D/following{/other_user}",
+        "gists_url": "https://api.github.com/users/github-actions%5Bbot%5D/gists{/gist_id}",
+        "starred_url": "https://api.github.com/users/github-actions%5Bbot%5D/starred{/owner}{/repo}",
+        "subscriptions_url": "https://api.github.com/users/github-actions%5Bbot%5D/subscriptions",
+        "organizations_url": "https://api.github.com/users/github-actions%5Bbot%5D/orgs",
+        "repos_url": "https://api.github.com/users/github-actions%5Bbot%5D/repos",
+        "events_url": "https://api.github.com/users/github-actions%5Bbot%5D/events{/privacy}",
+        "received_events_url": "https://api.github.com/users/github-actions%5Bbot%5D/received_events",
+        "type": "Bot",
+        "user_view_type": "public",
+        "site_admin": false
+      },
+      "content_type": "application/zip",
+      "state": "uploaded",
+      "size": 6447323,
+      "digest": null,
+      "download_count": 568,
+      "created_at": "2025-01-20T22:43:22Z",
+      "updated_at": "2025-01-20T22:43:22Z",
+      "browser_download_url": "https://github.com/zmkfirmware/zmk-studio/releases/download/v0.3.1/ZMK.Studio-0.3.1-1.x86_64.rpm"
+    },
+    {
+      "url": "https://api.github.com/repos/zmkfirmware/zmk-studio/releases/assets/221951337",
+      "id": 221951337,
+      "node_id": "RA_kwDOL8zQ3M4NOrVp",
+      "name": "ZMK.Studio_0.3.1_amd64.AppImage",
+      "label": "",
+      "uploader": {
+        "login": "github-actions[bot]",
+        "id": 41898282,
+        "node_id": "MDM6Qm90NDE4OTgyODI=",
+        "avatar_url": "https://avatars.githubusercontent.com/in/15368?v=4",
+        "gravatar_id": "",
+        "url": "https://api.github.com/users/github-actions%5Bbot%5D",
+        "html_url": "https://github.com/apps/github-actions",
+        "followers_url": "https://api.github.com/users/github-actions%5Bbot%5D/followers",
+        "following_url": "https://api.github.com/users/github-actions%5Bbot%5D/following{/other_user}",
+        "gists_url": "https://api.github.com/users/github-actions%5Bbot%5D/gists{/gist_id}",
+        "starred_url": "https://api.github.com/users/github-actions%5Bbot%5D/starred{/owner}{/repo}",
+        "subscriptions_url": "https://api.github.com/users/github-actions%5Bbot%5D/subscriptions",
+        "organizations_url": "https://api.github.com/users/github-actions%5Bbot%5D/orgs",
+        "repos_url": "https://api.github.com/users/github-actions%5Bbot%5D/repos",
+        "events_url": "https://api.github.com/users/github-actions%5Bbot%5D/events{/privacy}",
+        "received_events_url": "https://api.github.com/users/github-actions%5Bbot%5D/received_events",
+        "type": "Bot",
+        "user_view_type": "public",
+        "site_admin": false
+      },
+      "content_type": "application/zip",
+      "state": "uploaded",
+      "size": 84329664,
+      "digest": null,
+      "download_count": 4227,
+      "created_at": "2025-01-20T22:43:22Z",
+      "updated_at": "2025-01-20T22:43:24Z",
+      "browser_download_url": "https://github.com/zmkfirmware/zmk-studio/releases/download/v0.3.1/ZMK.Studio_0.3.1_amd64.AppImage"
+    },
+    {
+      "url": "https://api.github.com/repos/zmkfirmware/zmk-studio/releases/assets/221951335",
+      "id": 221951335,
+      "node_id": "RA_kwDOL8zQ3M4NOrVn",
+      "name": "ZMK.Studio_0.3.1_amd64.deb",
+      "label": "",
+      "uploader": {
+        "login": "github-actions[bot]",
+        "id": 41898282,
+        "node_id": "MDM6Qm90NDE4OTgyODI=",
+        "avatar_url": "https://avatars.githubusercontent.com/in/15368?v=4",
+        "gravatar_id": "",
+        "url": "https://api.github.com/users/github-actions%5Bbot%5D",
+        "html_url": "https://github.com/apps/github-actions",
+        "followers_url": "https://api.github.com/users/github-actions%5Bbot%5D/followers",
+        "following_url": "https://api.github.com/users/github-actions%5Bbot%5D/following{/other_user}",
+        "gists_url": "https://api.github.com/users/github-actions%5Bbot%5D/gists{/gist_id}",
+        "starred_url": "https://api.github.com/users/github-actions%5Bbot%5D/starred{/owner}{/repo}",
+        "subscriptions_url": "https://api.github.com/users/github-actions%5Bbot%5D/subscriptions",
+        "organizations_url": "https://api.github.com/users/github-actions%5Bbot%5D/orgs",
+        "repos_url": "https://api.github.com/users/github-actions%5Bbot%5D/repos",
+        "events_url": "https://api.github.com/users/github-actions%5Bbot%5D/events{/privacy}",
+        "received_events_url": "https://api.github.com/users/github-actions%5Bbot%5D/received_events",
+        "type": "Bot",
+        "user_view_type": "public",
+        "site_admin": false
+      },
+      "content_type": "application/zip",
+      "state": "uploaded",
+      "size": 6447010,
+      "digest": null,
+      "download_count": 3001,
+      "created_at": "2025-01-20T22:43:21Z",
+      "updated_at": "2025-01-20T22:43:21Z",
+      "browser_download_url": "https://github.com/zmkfirmware/zmk-studio/releases/download/v0.3.1/ZMK.Studio_0.3.1_amd64.deb"
+    },
+    {
+      "url": "https://api.github.com/repos/zmkfirmware/zmk-studio/releases/assets/221951339",
+      "id": 221951339,
+      "node_id": "RA_kwDOL8zQ3M4NOrVr",
+      "name": "ZMK.Studio_0.3.1_universal.dmg",
+      "label": "",
+      "uploader": {
+        "login": "github-actions[bot]",
+        "id": 41898282,
+        "node_id": "MDM6Qm90NDE4OTgyODI=",
+        "avatar_url": "https://avatars.githubusercontent.com/in/15368?v=4",
+        "gravatar_id": "",
+        "url": "https://api.github.com/users/github-actions%5Bbot%5D",
+        "html_url": "https://github.com/apps/github-actions",
+        "followers_url": "https://api.github.com/users/github-actions%5Bbot%5D/followers",
+        "following_url": "https://api.github.com/users/github-actions%5Bbot%5D/following{/other_user}",
+        "gists_url": "https://api.github.com/users/github-actions%5Bbot%5D/gists{/gist_id}",
+        "starred_url": "https://api.github.com/users/github-actions%5Bbot%5D/starred{/owner}{/repo}",
+        "subscriptions_url": "https://api.github.com/users/github-actions%5Bbot%5D/subscriptions",
+        "organizations_url": "https://api.github.com/users/github-actions%5Bbot%5D/orgs",
+        "repos_url": "https://api.github.com/users/github-actions%5Bbot%5D/repos",
+        "events_url": "https://api.github.com/users/github-actions%5Bbot%5D/events{/privacy}",
+        "received_events_url": "https://api.github.com/users/github-actions%5Bbot%5D/received_events",
+        "type": "Bot",
+        "user_view_type": "public",
+        "site_admin": false
+      },
+      "content_type": "application/zip",
+      "state": "uploaded",
+      "size": 12399976,
+      "digest": null,
+      "download_count": 6693,
+      "created_at": "2025-01-20T22:43:27Z",
+      "updated_at": "2025-01-20T22:43:28Z",
+      "browser_download_url": "https://github.com/zmkfirmware/zmk-studio/releases/download/v0.3.1/ZMK.Studio_0.3.1_universal.dmg"
+    },
+    {
+      "url": "https://api.github.com/repos/zmkfirmware/zmk-studio/releases/assets/221951388",
+      "id": 221951388,
+      "node_id": "RA_kwDOL8zQ3M4NOrWc",
+      "name": "ZMK.Studio_0.3.1_x64-setup.exe",
+      "label": "",
+      "uploader": {
+        "login": "github-actions[bot]",
+        "id": 41898282,
+        "node_id": "MDM6Qm90NDE4OTgyODI=",
+        "avatar_url": "https://avatars.githubusercontent.com/in/15368?v=4",
+        "gravatar_id": "",
+        "url": "https://api.github.com/users/github-actions%5Bbot%5D",
+        "html_url": "https://github.com/apps/github-actions",
+        "followers_url": "https://api.github.com/users/github-actions%5Bbot%5D/followers",
+        "following_url": "https://api.github.com/users/github-actions%5Bbot%5D/following{/other_user}",
+        "gists_url": "https://api.github.com/users/github-actions%5Bbot%5D/gists{/gist_id}",
+        "starred_url": "https://api.github.com/users/github-actions%5Bbot%5D/starred{/owner}{/repo}",
+        "subscriptions_url": "https://api.github.com/users/github-actions%5Bbot%5D/subscriptions",
+        "organizations_url": "https://api.github.com/users/github-actions%5Bbot%5D/orgs",
+        "repos_url": "https://api.github.com/users/github-actions%5Bbot%5D/repos",
+        "events_url": "https://api.github.com/users/github-actions%5Bbot%5D/events{/privacy}",
+        "received_events_url": "https://api.github.com/users/github-actions%5Bbot%5D/received_events",
+        "type": "Bot",
+        "user_view_type": "public",
+        "site_admin": false
+      },
+      "content_type": "application/zip",
+      "state": "uploaded",
+      "size": 4459240,
+      "digest": null,
+      "download_count": 6759,
+      "created_at": "2025-01-20T22:43:46Z",
+      "updated_at": "2025-01-20T22:43:47Z",
+      "browser_download_url": "https://github.com/zmkfirmware/zmk-studio/releases/download/v0.3.1/ZMK.Studio_0.3.1_x64-setup.exe"
+    },
+    {
+      "url": "https://api.github.com/repos/zmkfirmware/zmk-studio/releases/assets/221951386",
+      "id": 221951386,
+      "node_id": "RA_kwDOL8zQ3M4NOrWa",
+      "name": "ZMK.Studio_0.3.1_x64_en-US.msi",
+      "label": "",
+      "uploader": {
+        "login": "github-actions[bot]",
+        "id": 41898282,
+        "node_id": "MDM6Qm90NDE4OTgyODI=",
+        "avatar_url": "https://avatars.githubusercontent.com/in/15368?v=4",
+        "gravatar_id": "",
+        "url": "https://api.github.com/users/github-actions%5Bbot%5D",
+        "html_url": "https://github.com/apps/github-actions",
+        "followers_url": "https://api.github.com/users/github-actions%5Bbot%5D/followers",
+        "following_url": "https://api.github.com/users/github-actions%5Bbot%5D/following{/other_user}",
+        "gists_url": "https://api.github.com/users/github-actions%5Bbot%5D/gists{/gist_id}",
+        "starred_url": "https://api.github.com/users/github-actions%5Bbot%5D/starred{/owner}{/repo}",
+        "subscriptions_url": "https://api.github.com/users/github-actions%5Bbot%5D/subscriptions",
+        "organizations_url": "https://api.github.com/users/github-actions%5Bbot%5D/orgs",
+        "repos_url": "https://api.github.com/users/github-actions%5Bbot%5D/repos",
+        "events_url": "https://api.github.com/users/github-actions%5Bbot%5D/events{/privacy}",
+        "received_events_url": "https://api.github.com/users/github-actions%5Bbot%5D/received_events",
+        "type": "Bot",
+        "user_view_type": "public",
+        "site_admin": false
+      },
+      "content_type": "application/zip",
+      "state": "uploaded",
+      "size": 5636096,
+      "digest": null,
+      "download_count": 1174,
+      "created_at": "2025-01-20T22:43:46Z",
+      "updated_at": "2025-01-20T22:43:46Z",
+      "browser_download_url": "https://github.com/zmkfirmware/zmk-studio/releases/download/v0.3.1/ZMK.Studio_0.3.1_x64_en-US.msi"
+    },
+    {
+      "url": "https://api.github.com/repos/zmkfirmware/zmk-studio/releases/assets/221951343",
+      "id": 221951343,
+      "node_id": "RA_kwDOL8zQ3M4NOrVv",
+      "name": "ZMK.Studio_universal.app.tar.gz",
+      "label": "",
+      "uploader": {
+        "login": "github-actions[bot]",
+        "id": 41898282,
+        "node_id": "MDM6Qm90NDE4OTgyODI=",
+        "avatar_url": "https://avatars.githubusercontent.com/in/15368?v=4",
+        "gravatar_id": "",
+        "url": "https://api.github.com/users/github-actions%5Bbot%5D",
+        "html_url": "https://github.com/apps/github-actions",
+        "followers_url": "https://api.github.com/users/github-actions%5Bbot%5D/followers",
+        "following_url": "https://api.github.com/users/github-actions%5Bbot%5D/following{/other_user}",
+        "gists_url": "https://api.github.com/users/github-actions%5Bbot%5D/gists{/gist_id}",
+        "starred_url": "https://api.github.com/users/github-actions%5Bbot%5D/starred{/owner}{/repo}",
+        "subscriptions_url": "https://api.github.com/users/github-actions%5Bbot%5D/subscriptions",
+        "organizations_url": "https://api.github.com/users/github-actions%5Bbot%5D/orgs",
+        "repos_url": "https://api.github.com/users/github-actions%5Bbot%5D/repos",
+        "events_url": "https://api.github.com/users/github-actions%5Bbot%5D/events{/privacy}",
+        "received_events_url": "https://api.github.com/users/github-actions%5Bbot%5D/received_events",
+        "type": "Bot",
+        "user_view_type": "public",
+        "site_admin": false
+      },
+      "content_type": "application/zip",
+      "state": "uploaded",
+      "size": 11766995,
+      "digest": null,
+      "download_count": 167,
+      "created_at": "2025-01-20T22:43:28Z",
+      "updated_at": "2025-01-20T22:43:29Z",
+      "browser_download_url": "https://github.com/zmkfirmware/zmk-studio/releases/download/v0.3.1/ZMK.Studio_universal.app.tar.gz"
+    }
+  ],
+  "tarball_url": "https://api.github.com/repos/zmkfirmware/zmk-studio/tarball/v0.3.1",
+  "zipball_url": "https://api.github.com/repos/zmkfirmware/zmk-studio/zipball/v0.3.1",
+  "body": "## [0.3.1](https://github.com/zmkfirmware/zmk-studio/compare/v0.3.0...v0.3.1) (2025-01-20)\n\n\n### Bug Fixes\n\n* **ci:** Support token auth on release fetch ([#126](https://github.com/zmkfirmware/zmk-studio/issues/126)) ([0662e6d](https://github.com/zmkfirmware/zmk-studio/commit/0662e6d506715232b1cee872f7354435c0cdea16))",
+  "reactions": {
+    "url": "https://api.github.com/repos/zmkfirmware/zmk-studio/releases/195729513/reactions",
+    "total_count": 10,
+    "+1": 0,
+    "-1": 0,
+    "laugh": 0,
+    "hooray": 0,
+    "confused": 0,
+    "heart": 10,
+    "rocket": 0,
+    "eyes": 0
+  }
+}
-- 
2.51.1

